app.ts

```diff
@@ -244,41 +246,43 @@ export class App<
       })
     }
     mw.push({ type: 'mw', handler: this.notFound, path: '/' })
-
     const handle =
       (mw: Middleware<Req, Res>) =>
       async (req: Req, res: Res, next: NextFunction) => {
         const { handler, type, pattern } = mw
-
         const params =
           type === 'route' && pattern?.exec(req.url)?.pathname.groups || {}
 
         req.params = params as Record<string, string>
-
+        if (!req.path) req.path = req._urlObject.pathname
         if (this.settings?.enableReqRoute) {
           req.route = getRouteFromApp(this.middleware as any, handler as any)
         }
         await applyHandler<Req, Res>(handler)(req, res, next)
       }
 
-    let idx = 0, err: unknown | undefined
+    let idx = 0, err: unknown | undefined = null
     const next: NextFunction = async (error) => {
-      if (error) err = error
+      if (error) {
+        err = error
+      }
       return await loop()
     }
     const loop = async () =>
       idx < mw.length
-        ? await handle(mw[idx++])(
-          req,
-          res as Res,
-          next,
-        )
+        ? await handle(mw[idx++])(req, res as Res, next)
         : undefined
 
     await loop()
 
-    if (err) throw err
+    if (err instanceof Response) throw err
+    else if (err) {
+      if (this[hasSetCustomErrorHandler]) throw await this.onError(err, req)
+      else throw err
+    }
+    throw new Response(res._body, res._init)
```

Comment:
why do you throw a response?

Reply:
I need to persist the response. If I don't throw, the response gets lost in the function stack during recursion.